src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -739,6 +739,84 @@ class NBDWatchCtx : public librbd::UpdateWatchCtx
   librados::IoCtx &io_ctx;
   librbd::Image &image;
   unsigned long size;
+  ceph::condition_variable cond;
+  ceph::mutex lock = ceph::make_mutex("NBDWatchCtx::Locker");
+
+  class ThreadHelper : public Thread
+  {
+  public:
+    typedef void (NBDWatchCtx::*entry_func)();
+  private:
+    NBDWatchCtx &server;
```

Comment:
When cutting and pasting, at least rename the variables ;)

Reply:
On a more serious note, is this `ThreadHelper` business actually needed?  In `NBDServer` it serves three different threads (entry funcs), thus avoiding code duplication.  Here, `NBDWatchCtx` has a single thread so the thread can be hard coded.

Reply:
Sure